tools/ldb_cmd.cc

```diff
@@ -114,11 +114,83 @@ const std::string LDBCommand::ARG_READ_TIMESTAMP = "read_timestamp";
 const char* LDBCommand::DELIM = " ==> ";
 
 namespace {
+// Helper class to iterate WAL logs in a directory in chronological order.
+class WALPicker {
+ public:
+  explicit WALPicker(const std::string& parent_dir,
+                     const std::vector<std::string>& filenames);
+  // REQUIRES Valid() == true
+  std::string GetNextWAL();
+  bool Valid() const { return wal_file_iter_ != log_files_.end(); }
+
+ private:
+  // WAL log file names(s)
+  std::string parent_dir_;
+  std::vector<std::string> log_files_;
+  std::vector<std::string>::const_iterator wal_file_iter_;
+};
+
+WALPicker::WALPicker(const std::string& parent_dir,
+                     const std::vector<std::string>& filenames)
+    : parent_dir_(parent_dir) {
+  // populate wal logs
+  assert(!filenames.empty());
+  for (const auto& fname : filenames) {
+    uint64_t file_num = 0;
+    FileType file_type;
+    bool parse_ok = ParseFileName(fname, &file_num, &file_type);
+    if (parse_ok && file_type == kWalFile) {
+      log_files_.push_back(fname);
+    }
+  }
+
+  std::sort(log_files_.begin(), log_files_.end(),
+            [](const std::string& lhs, const std::string& rhs) {
+              uint64_t num1 = 0;
+              uint64_t num2 = 0;
+              FileType type1;
+              FileType type2;
+              bool parse_ok1 = ParseFileName(lhs, &num1, &type1);
+              bool parse_ok2 = ParseFileName(rhs, &num2, &type2);
+#ifndef NDEBUG
+              assert(parse_ok1);
+              assert(parse_ok2);
+#else
+        (void)parse_ok1;
+        (void)parse_ok2;
+#endif
+              return num1 < num2;
+            });
+  wal_file_iter_ = log_files_.begin();
+}
 
-void DumpWalFile(Options options, std::string wal_file, bool print_header,
-                 bool print_values, bool is_write_committed,
+std::string WALPicker::GetNextWAL() {
+  assert(Valid());
+  std::string ret;
+  if (wal_file_iter_ != log_files_.end()) {
```

Comment:
This check seems redundant, since we `assert(Valid());` above

Reply:
I think this check is helpful if that assert line is not compiled in the build.